pkg/resourceinterpreter/default/thirdparty/thirdparty_test.go

```diff
@@ -0,0 +1,187 @@
+		err = decoder.Decode(statusItem)
+		if err != nil {
+			break
+		}
+		statusItems = append(statusItems, *statusItem)
+	}
+	if err != io.EOF {
+		t.Fatal(err)
+	}
+
+	return statusItems
+}
+
+type TestStructure struct {
+	Tests []IndividualTest `yaml:"tests"`
+}
+
+type IndividualTest struct {
+	DesiredInputPath  string `yaml:"desiredInputPath,omitempty"`
+	ObservedInputPath string `yaml:"observedInputPath,omitempty"`
+	StatusInputPath   string `yaml:"statusInputPath,omitempty"`
+	DesiredReplica    int64  `yaml:"desiredReplicas,omitempty"`
+	Operation         string `yaml:"operation"`
+}
+
+func checkInterpretationRule(t *testing.T, path string, configs []*configv1alpha1.ResourceInterpreterCustomization) {
+	ipt := declarative.NewConfigurableInterpreter(nil)
+	ipt.LoadConfig(configs)
+
+	dir := filepath.Dir(path)
+	yamlBytes, err := os.ReadFile(dir + "/customizations_tests.yaml")
+	if err != nil {
+		t.Fatal(err)
+	}
+	var resourceTest TestStructure
+	err = yaml.Unmarshal(yamlBytes, &resourceTest)
+	if err != nil {
+		t.Fatal(err)
+	}
+	for _, customization := range configs {
+		for _, input := range resourceTest.Tests {
+			rule := rules.GetByOperation(input.Operation)
+			if rule == nil {
+				t.Fatalf("operation %s is not supported. Use one of: %s", input.Operation, strings.Join(rules.Names(), ", "))
+			}
+			err = checkScrip(rule.GetScript(customization))
+			if err != nil {
+				t.Fatalf("checking %s of %s, expected nil, but got: %v", rule.Name(), customization.Name, err)
+			}
+			args := interpreter.RuleArgs{Replica: input.DesiredReplica}
+			if input.DesiredInputPath != "" {
+				args.Desired = getObj(t, dir+"/"+strings.TrimPrefix(input.DesiredInputPath, "/"))
+			}
+			if input.ObservedInputPath != "" {
+				args.Observed = getObj(t, dir+"/"+strings.TrimPrefix(input.ObservedInputPath, "/"))
+			}
+			if input.StatusInputPath != "" {
+				args.Status = getAggregatedStatusItems(t, dir+"/"+strings.TrimPrefix(input.StatusInputPath, "/"))
+			}
+			if result := rule.Run(ipt, args); result.Err != nil {
+				t.Fatalf("execute %s %s error: %v\n", customization.Name, rule.Name(), result.Err)
+			}
+		}
+	}
+}
+
+func TestThirdPartyCustomizationsFile(t *testing.T) {
+	err := filepath.Walk("resourcecustomizations", func(path string, f os.FileInfo, err error) error {
+		if err != nil {
+			// cannot happen
+			return err
+		}
+		if f.IsDir() {
+			return nil
+		}
+		if strings.Contains(path, "testdata") {
+			return nil
+		}
+		if filepath.Base(path) != configurableInterpreterFile {
+			return nil
+		}
+
+		data, err := os.ReadFile(path)
+		if err != nil {
+			// cannot happen
+			return err
+		}
+		var configs []*configv1alpha1.ResourceInterpreterCustomization
+		decoder := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(data), 4096)
+		for {
+			config := &configv1alpha1.ResourceInterpreterCustomization{}
+			err = decoder.Decode(config)
+			if err != nil {
+				break
+			}
+			dirSplit := strings.Split(path, "/")
```

Comment:
I have a small question: do we need to consider testing under windows?
```go
sysType := runtime.GOOS
var separator string
if sysType == "linux" {
	separator = "/"
}
if sysType == "windows" {
	separator = "\\"
}
```

Reply:
Done.